subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultSourceDirectorySet.java

```diff
@@ -57,10 +56,7 @@
     private final FileCollection dirs;
     private final Property<File> outputDir;
 
-    // Note: don't actually remove this in 6.0, the deprecation is here to encourage people to use ObjectFactory instead. Just remove the overload and the nag and leave the method here
-    @Deprecated
```

Comment:
Should we leave the constructor deprecated (no nag) so plugin devs will get IDE hints that they shouldn't use?

Reply:
It says "Just remove the overload" in the comment. But since there is no overload, I assume it was meant to read the annotation/deprecated.

I'll merge this and leave it up to @adammurdoch to re-add the `@Deprecated` if he thinks it should be there.